staging/src/k8s.io/component-base/zpages/statusz/registry.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statusz
+
+import (
+	"time"
+
+	"github.com/blang/semver/v4"
+	"k8s.io/component-base/version"
+	"k8s.io/klog/v2"
+
+	"k8s.io/component-base/featuregate"
+	compbasemetrics "k8s.io/component-base/metrics"
+)
+
+type statuszRegistry interface {
+	processStartTime() time.Time
+	goVersion() string
+	binaryVersion() semver.Version
+	emulationVersion() semver.Version
+	minCompatibilityVersion() semver.Version
+}
+
+type registry struct{}
+
+func (registry) processStartTime() time.Time {
+	start, err := compbasemetrics.GetProcessStart()
+	if err != nil {
+		klog.Errorf("Could not get process start time, %v", err)
+	}
+
+	return time.Unix(int64(start), 0)
+}
+
+func (registry) goVersion() string {
+	return version.Get().GoVersion
+}
+
+func (registry) binaryVersion() semver.Version {
+	var binaryVersion semver.Version
+	binaryVersion, err := semver.ParseTolerant(featuregate.DefaultComponentGlobalsRegistry.EffectiveVersionFor(featuregate.DefaultKubeComponent).BinaryVersion().String())
+	if err != nil {
+		klog.Errorf("err parsing binary version: %v", err)
+	}
+
+	return binaryVersion
+}
+
+func (registry) emulationVersion() semver.Version {
+	var emulationVersion semver.Version
+	emulationVersion, err := semver.ParseTolerant(featuregate.DefaultComponentGlobalsRegistry.EffectiveVersionFor(featuregate.DefaultKubeComponent).EmulationVersion().String())
+	if err != nil {
+		klog.Errorf("err parsing emulationVersion version: %v", err)
+	}
+
+	return emulationVersion
+}
+
+func (registry) minCompatibilityVersion() semver.Version {
```

Comment:
This may be removed in the near future, it's probably fine to keep it for now though. binary and emulation version should be the most important for compat version info.

Reply:
Removed min compat version from the response now.